DeviceCode/pal/lwip_1_4_1_os/SocketsDriver/lwIP__Sockets.cpp

```diff
@@ -562,13 +562,18 @@ int LWIP_SOCKETS_Driver::SetSockOpt( SOCK_SOCKET socket, int level, int optname,
             nativeOptionName = GetNativeSockOption(optname);            
 
             switch(optname)
-            {
+            {        
+                // LINGER and DONTLINGER are not implemented in LWIP
+                case SOCK_SOCKO_LINGER:
+                case SOCK_SOCKO_DONTLINGER:
+                    errno = SOCK_ENOPROTOOPT;
+                    return SOCK_SOCKET_ERROR;

```

Comment:
Any chance we could use a more descriptive error code! This smells a bit like E_FAIL... (same for other cases in here)